src/Sentry.Unity/UnityEventProcessor.cs

```diff
@@ -173,28 +181,64 @@ static string ToGraphicShaderLevelDescription(int shaderLevel)
 
         private void PopulateUnity(Protocol.Unity unity)
         {
-            unity.InstallMode = _application.InstallMode.ToString();
+            unity.InstallMode = _mainThreadData.InstallMode;
         }
 
         private void PopulateTags(SentryEvent @event)
         {
-            @event.SetTag("unity.install_mode", _application.InstallMode.ToString());
+            if (_mainThreadData.InstallMode is not null)
+            {
+                @event.SetTag("unity.install_mode", _mainThreadData.InstallMode);
+            }
 
             if (_mainThreadData.SupportsDrawCallInstancing.HasValue)
             {
                 @event.SetTag("unity.gpu.supports_instancing", _mainThreadData.SupportsDrawCallInstancing.Value ? "true" : "false");
             }
 
-            if (_mainThreadData.DeviceType is not null)
+            if (_mainThreadData.DeviceType is not null && _mainThreadData.DeviceType.IsValueCreated)
             {
-                @event.SetTag("unity.device.device_type", _mainThreadData.DeviceType);
+                @event.SetTag("unity.device.device_type", _mainThreadData.DeviceType.Value);
             }
 
-            if (_sentryOptions.SendDefaultPii && _mainThreadData.DeviceUniqueIdentifier is not null)
+            if (_sentryOptions.SendDefaultPii && _mainThreadData.DeviceUniqueIdentifier is not null && _mainThreadData.DeviceUniqueIdentifier.IsValueCreated)
             {
-                @event.SetTag("unity.device.unique_identifier", _mainThreadData.DeviceUniqueIdentifier);
+                @event.SetTag("unity.device.unique_identifier", _mainThreadData.DeviceUniqueIdentifier.Value);
             }
         }
+
+        /// <summary>
+        /// - If UI thread, extract the value (can be null)
+        /// - If non-UI thread, check if value is created, then extract
+        /// - 'null' otherwise
+        /// </summary>
+        private string? SafeLazyUnwrap(Lazy<string>? lazyValue)
+        {
+            if (lazyValue == null)
+            {
+                return null;
+            }
+
+            return _mainThreadData.IsMainThread()
+                ? lazyValue.Value
+                : lazyValue.IsValueCreated
+                    ? lazyValue.Value
+                    : null;
```

Comment:
Log a debug message that because we're not in the UI thread and value hasn't be unwrapped yet, we're returning null for this property: